### PR TITLE
Sean/add annotations

### PR DESCRIPTION
**Pull request by Sean-Holcomb**
Tested by filtering on annotations using requests like this on bolt-1:

/model/aggregatedCostModel?window=1d&offset=1m&aggregation=namespace&allocateIdle=true&timeSeries=true&annotations=test_kubecost_annotation_instance%3Dkubecost&efficiency=true&aggregationSubfield=&sharedNamespaces=&sharedSplit=weighted&sharedLabelNames=&sharedLabelValues=&sharedSplit=weighted&req=635724

Additional annotations to filter on can be found in the bolt-1 prometheus dashboard by querying kube_pod_annotations

### CODE REVIEW

**Comment by dwbrown2** [comment]
Nice! Does this endpoint now support `aggregation=annotation`? 

Can we do a quick doc on how to enable these + metrics plus how to use them?

**Comment by Sean-Holcomb** [comment]
Thanks @mbolt35 for multi-filter bug fix

**Comment by Sean-Holcomb** [comment]
> Nice! Does this endpoint now support `aggregation=annotation`?
> 
> Can we do a quick doc on how to enable these + metrics plus how to use them?
Pr for aggregation on annotations
https://github.com/kubecost/cost-model/pull/657

Working with Ajay on adding in documentation

**Review by nikovacevic — CHANGES_REQUESTED** [review]
Good work on this so far. I haven't manually tested yet, but if we address the few small things I've noted and get it thoroughly tested, then I think this will be ready to approve and merge. Let me know if you have any questions.

File: pkg/costmodel/aggregation.go
Comment (Sean-Holcomb) [reply pkg/costmodel/aggregation.go:1848]: label sanitization happens later in the process and these particular snippet doesn't account for multiple label filters that are comma separated like the later one does

**Comment by AjayTripathy** [comment]
This LGTM, but I'll let @nikovacevic give the final review.

File: pkg/costmodel/aggregation.go
Comment (mbolt35) [reply pkg/costmodel/aggregation.go:1864]: @nikovacevic Yeah, this code is completely unnecessary.

File: pkg/costmodel/costmodel.go
Comment (mbolt35) [inline pkg/costmodel/costmodel.go:88]: `json:"annotations,omitempty"` camel case for json properties

**Review by mbolt35 — APPROVED** [review]
LGTM after the one small change to the annotations json property.

File: pkg/costmodel/costmodel.go
Comment (Sean-Holcomb) [reply pkg/costmodel/costmodel.go]: updated

File: pkg/costmodel/costmodel.go
Comment (mbolt35) [inline pkg/costmodel/costmodel.go:1737]: We can probably rename `appendNamespaceLabels` to just `mergeStringMap` or something :)

**Comment by Sean-Holcomb** [comment]
This PR is a subset of this other pr

kubecost/cost-model#657

**Pull request closed without merging**
